scalafix-internal/rules/src/main/scala-2/fix/UpstreamLibs.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fix
+
+import scalafix.v1._
+
+import scala.meta._
+
+class UpstreamLibs extends SemanticRule("Http4sUpstreamLibs") {
```

Comment:
These are not always equivalent, because `as(a)` is strict and `map(_ => a)` is lazy.  We had an optimization around that recently, though I don't remember it being on `Stream`.

Why target `Stream` specifically as opposed to other functors?  Just have to start somewhere?  I think the cost is one-per-stream instead of one-per-output, which helps.

Reply:
Yeah, it was really just a low-hanging bike shed to start hacking on. Happy to remove that one.

Reply:
I'm inclined toward no on this one, but yes to the rest.